org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/launch/GradleRunConfigurationAttributes.java

```diff
@@ -197,13 +212,29 @@ public boolean hasSameUniqueAttributes(ILaunchConfiguration launchConfiguration)
     }
 
     public FixedRequestAttributes toFixedRequestAttributes() {
-        return FixedRequestAttributesBuilder.fromWorkspaceSettings(getWorkingDir())
-            .gradleUserHome(getGradleUserHome())
-            .gradleDistribution(getGradleDistribution())
-            .javaHome(getJavaHome())
-            .jvmArguments(getJvmArguments())
-            .arguments(getArguments())
-            .build();
+        if (isOverrideWorkspaceSettings()) {
```

Comment:
It was a good idea. I changed the implementation and the code looks much cleaner now.